InteroperabilityTest_Engine/Query/HasMatchingIds.cs

```diff
@@ -0,0 +1,95 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Test.Results;
+using BH.oM.Test.Interoperability.Results;
+using BH.oM.Test;
+
+namespace BH.Engine.Test.Interoperability
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks if the two TestResult has the same ID. Used for comparing two test results with each other.")]
+        [Input("a", "First TestResult to compare.")]
+        [Input("b", "Second TestResult to compare.")]
+        [Output("sameId", "Returns true if the two TestResult have the same ID.")]
+        public static bool HasMatchingIds(this TestResult a, TestResult b)
```

Comment:
Should be `IDs` to be consistent - haven't suggested as a change cause it probably needs updating in lots of other places. But same as properties.